src/nbl/asset/material_compiler/CMaterialCompilerGLSLBackendCommon.cpp

```diff
@@ -1303,29 +1334,48 @@ auto CMaterialCompilerGLSLBackendCommon::compile(SContext* _ctx, IR* _ir, E_GENE
 			res.bsdfData.push_back(bsdf_data);
 		}
 
-		result_t::instr_streams_t streams;
+		oriented_material_t material;
 		{
-			streams.offset = res.instructions.size();
+			material.instr_offset = res.instructions.size();
 
-			streams.rem_and_pdf_count = rem_pdf_stream.size();
+			material.rem_pdf_count = rem_pdf_stream.size();
 			res.instructions.insert(res.instructions.end(), rem_pdf_stream.begin(), rem_pdf_stream.end());
 
-			streams.gen_choice_count = gen_choice_stream.size();
+			material.genchoice_count = gen_choice_stream.size();
 			res.instructions.insert(res.instructions.end(), gen_choice_stream.begin(), gen_choice_stream.end());
 
-			streams.norm_precomp_count = normal_precomp_stream.size();
+			material.nprecomp_count = normal_precomp_stream.size();
 			res.instructions.insert(res.instructions.end(), normal_precomp_stream.begin(), normal_precomp_stream.end());
 
-			streams.prefetch_offset = res.prefetch_stream.size();
-			streams.tex_prefetch_count = tex_prefetch_stream.size();
+			material.prefetch_offset = res.prefetch_stream.size();
+			material.prefetch_count = tex_prefetch_stream.size();
 			res.prefetch_stream.insert(res.prefetch_stream.end(), tex_prefetch_stream.begin(), tex_prefetch_stream.end());
 		}
 
-		res.streams.insert({root,streams});
-
-		res.noNormPrecompStream = res.noNormPrecompStream && (streams.norm_precomp_count==0u);
-		res.noPrefetchStream = res.noPrefetchStream && (streams.tex_prefetch_count==0u);
+		res.noNormPrecompStream = res.noNormPrecompStream && (material.nprecomp_count==0u);
+		res.noPrefetchStream = res.noPrefetchStream && (material.prefetch_count==0u);
 		res.usedRegisterCount = std::max(res.usedRegisterCount, instr_stream::MAX_REGISTER_COUNT-remainingRegisters);
+
+		compiledBSDFRootNodes.emplace(root, material);
+
+		return material;
+	};
+
+	for (const IR::INode* root : _ir->roots)
+	{
+		oriented_material_t material;
+		if (root->symbol == IR::INode::ES_EMISSION) {
+			auto* node = static_cast<const IR::CEmissionNode*>(root);
+			material = compileBSDFRootNode(node->children[0]);
+			material.emitter_id = getOrCreateEmitter(node->emitter);
+		}
+		else {
+			material = compileBSDFRootNode(root);
+			material.emitter_id = 0;
+		}
+
+		printf("asdfasf %d\n", material.emitter_id);
```

Comment:
we have a dedicated log for logging out the material compiler stuff, ex 22 produces a log.txt everytime with a summary of everything

Reply:
please print to my debug log